x/incentive/README.md

```diff
@@ -0,0 +1,301 @@
+providers, the [`x/distribution`](https://docs.cosmos.network/main/build/modules/distribution)
+module will then distribute the rest to native stakers and validators.
+
+## 2. States
+
+The Checkpointing module maintains the following KV stores.
+
+### Prefixes
+
+- `DelegatorWithdrawAddrPrefix`: Used for storing the withdraw address for each
+    delegator.
+- `RefundableMsgKeySetPrefix`: Used for storing the set of hashes of messages
+    that can be refunded.
+- `FinalityProviderCurrentRewardsKeyPrefix`: Used for storing the Current
+    rewards of finality provider by address.
+- `BTCDelegationRewardsTrackerKeyPrefix`: Used for for BTC delegation rewards
+    tracker info
+- `FinalityProviderHistoricalRewardsKeyPrefix`: Used for storing the Historical
+    rewards of finality provider by address and period.
+
+### Keys
+
+- `ParamsKey`: Used for storing the parameters of the Incentive module.
+- `BTCStakingGaugeKey`: Used for storing the BTC staking gauge at each block
+    height.
+- `RewardGaugeKey`: Used for storing the reward gauge for a given stakeholder
+    in a given type.
+- `BTCDelegatorToFPKey`: Used for storing the map reference from a delegator
+    to a finality provider.
+
+### 2.1. Parameters
+
+The [parameter management](./keeper/params.go) maintains the
+incentive module's parameters. The BTC Checkpoint module's
+parameters are represented as a `Params` object
+[here](../proto/babylon/incentive/params.proto)
+defined as follows:
+
+```protobuf
+message Params {
+    option (gogoproto.goproto_stringer) = false;
+    string btc_staking_portion = 1 [
+        (cosmos_proto.scalar) = "cosmos.Dec",
+        (gogoproto.customtype) = "cosmossdk.io/math.LegacyDec",
+        (gogoproto.nullable) = false
+    ];
+}
+```
+
+### 2.2. Gauge
+
+This object stores rewards that are to be distributed, It can hold multiple
+denominations of coins and is managed by
+[reward gauge management](./keeper/btc_staking_gauge.go).
+
+```protobuf
+message Gauge {
+    repeated cosmos.base.v1beta1.Coin coins = 1 [
+        (gogoproto.nullable) = false,
+        (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+    ];
+}
+```
+
+### 2.3. Reward Gauge
+
+The `RewardGauge` is  managed by [reward gauge management](./keeper/reward_gauge.go)
+and is used to track rewards distributed to a BTC staking stakeholder. It
+includes both the accumulated rewards and the withdrawn rewards, allowing the
+ system to manage the lifecycle of rewards for each stakeholder.
+
+```protobuf
+message RewardGauge {
+repeated cosmos.base.v1beta1.Coin coins = 1 [
+        (gogoproto.nullable) = false,
+        (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+    ];
+    repeated cosmos.base.v1beta1.Coin withdrawn_coins = 2 [
+        (gogoproto.nullable) = false,
+        (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+    ];
+}
+```
+
+### 2.4. FinalityProviderHistoricalRewards
```

Comment:
for this it might be important to note the key used to store this structure 
which in this case is the fp baby address and the **period** 

https://github.com/babylonlabs-io/babylon/blob/cccef04aa4e6b7f54141735235bea3cb04b8890f/x/incentive/keeper/reward_tracker_store.go#L147